test/unit/printer/sourcemaps.spec.ts

```diff
@@ -326,3 +327,28 @@ test("Inline sourcemaps", () => {
     const inlineSourceMap = Buffer.from(inlineSourceMapMatch, "base64").toString();
     expect(inlineSourceMap).toBe(file.luaSourceMap);
 });
+
+test("loadstring sourceMapTraceback gives traceback", () => {
+    const strCodeBuilder = util.testModule`
+        function bar() {
+            const trace = (debug.traceback as (this: void)=>string)();
+            return trace;
+        }
+        return bar();
+    `.setOptions({ sourceMapTraceback: true, luaTarget: LuaTarget.Lua51 });
+
+    const strCode = strCodeBuilder.getMainLuaCodeChunk();
+    const file = `
+        const luaCode: string = \`${strCode}\`;
+        return (loadstring as (this: void, string: string, chunkname?: string) => LuaMultiReturn<[() => any] | [undefined, string]>)
+            (luaCode, "foo.lua")()
+    `;
+    const builder = util.testModule`
+        return (require as (this: void, modname: string) => any)("foo");
+    `
+        .addExtraFile("foo.ts", file)
+        .setOptions({ sourceMapTraceback: true, luaTarget: LuaTarget.Lua51 });
+
+    const traceback = builder.getLuaExecutionResult();
+    expect(traceback).toContain("foo.ts");
+});
```

Comment:
This looks way too complicated, couldn't you just do something like:
```ts
    const builder = util.testFunction`
        return loadstring(
            "return debug.traceback();",
            "foo.lua"
        )();
    `
        .setTsHeader("declare function loadstring(this: void, str: string, chunkName: string): () => unknown;")
        .setOptions({ sourceMapTraceback: true, luaTarget: LuaTarget.Lua51 });

    const traceback = builder.getLuaExecutionResult();
    expect(traceback).toContain("foo.ts");
```

Reply:
> This looks way too complicated, couldn't you just do something like:
> 
> ```ts
>     const builder = util.testFunction`
>         return loadstring(
>             "return debug.traceback();",
>             "foo.lua"
>         )();
>     `
>         .setTsHeader("declare function loadstring(this: void, str: string, chunkName: string): () => unknown;")
>         .setOptions({ sourceMapTraceback: true, luaTarget: LuaTarget.Lua51 });
> 
>     const traceback = builder.getLuaExecutionResult();
>     expect(traceback).toContain("foo.ts");
> ```

This is intentional.
The compiled lua code and sourcemap info should be passed into the loadstring together , otherelse the foo.lua could not be mapped back to foo.ts .

Reply:
I played around with it a bit, a much nicer test seems to be:
```ts
const loadStrCode = transpileString(
        `function bar() {
            const trace = (debug.traceback as (this: void)=>string)();
            return trace;
        }
        return bar();`,
        { sourceMapTraceback: true, luaTarget: LuaTarget.Lua51 }
    ).file?.lua;

    const builder = util.testModule`
        const luaCode = \`${loadStrCode}\`;
        return loadstring(luaCode, "foo.lua")();
    `
        .setTsHeader("declare function loadstring(this: void, string: string, chunkname: string): () => unknown")
        .setOptions({ sourceMapTraceback: true, luaTarget: LuaTarget.Lua51 });

    const traceback = builder.getLuaExecutionResult();
    expect(traceback).toContain("foo.ts");
```
Seems to work and is much neater.

Reply:
Cool. I will commit with your nicer test.